bin/sensei.js

```diff
@@ -225,6 +226,42 @@ cli(
     })
     .demandCommand(1, 1, "One command must be specified")
     .strict()
+    .middleware([
+      function handleConfig(argv, yargs) {
+        let configFilePath = argv.config;
+        if (!path.isAbsolute(configFilePath)) {
+          configFilePath = path.join(argv.material, configFilePath);
+        }
+        let config = {};
+        try {
+          config = JSON.parse(fs.readFileSync(configFilePath, "utf-8"));
+        } catch (err) {
+          if (
+            yargs.parsed.defaulted?.config == undefined ||
+            fs.existsSync(configFilePath)
+          ) {
+            console.warn(
+              `Unable to read config file '${configFilePath}' (${err.message}), skipping config.`
+            );
+          }
+        }
```

Comment:
I would separate the different cases:
- the file does not exist and `--config` was not defined explicitly => config defaults to empty object
- the file does not exist and `--config` was defined explicitly => error
- the file exists but cannot be read => error
- the file exists, can be read, but cannot be parsed as json => error

```suggestion
        let config = {};
        try {
          config = fs.readFileSync(configFilePath).toString()
        } catch (err) {
          if (err.code === 'ENOENT' && !yargs.parsed.defaulted?.config) {
            config = "{}"
          } else {
            throw Object.assign(new Error(`Cannot read config file '${configFilePath}': ${err.message}`), err)
          }
        }
        try {
          config = JSON.parse(config);
        } catch (err) {
          throw new Error(`Cannot parse the content of config file '${configFilePath}' as JSON: ${err.message}`)
        }
```

Reply:
Sounds good.
I integrated your suggestion while fixing formatting and adapting logic (any reading file error should be raised, but only when using explicit `--config` option).
I also added an info message about loading config from file (without error).